src/engine.ml

```diff
@@ -881,14 +886,19 @@ let expected_packet session transport data =
      both from their side, and acks from our side *)
   let* () =
     guard
-      (Int32.equal session.their_replay_id hdr.Packet.replay_id)
-      (`Non_monotonic_replay_id (transport, hdr))
+      (Int32.unsigned_compare session.their_replay_id hdr.Packet.replay_id <= 0)
```

Comment:
this actually changes behaviour, but imho this is correct -- there may be lost packets, and all we care is that the replay_id increases (so it should be greater or equal the one we expect)